app/models/Amounts.scala

```diff
@@ -1,48 +1,51 @@
 package models
 
-import io.circe.{ Decoder, Encoder }
-import io.circe.generic.auto._
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.auto._
+import io.circe.generic.extras.semiauto._
+import io.circe.{Decoder, Encoder}
 
-case class AmountsSelection(
+case class AmountValuesObject(
     amounts: List[Int], 
     defaultAmount: Int,
-    hideChooseYourAmount: Option[Boolean],
+    hideChooseYourAmount: Boolean,
 )
 
-case class ContributionAmounts(
-    ONE_OFF: AmountsSelection,
-    MONTHLY: AmountsSelection,
-    ANNUAL: AmountsSelection,
+case class AmountsCardData(
+    ONE_OFF: AmountValuesObject,
+    MONTHLY: AmountValuesObject,
+    ANNUAL: AmountValuesObject,
 )
 
-case class AmountsTestVariant(
-    name: String,
-    amounts: ContributionAmounts,
+case class AmountsVariant(
+  variantName: String,
+  defaultContributionType: String,
+  displayContributionType: List[String],
+  amountsCardData: AmountsCardData,
 )
 
 case class AmountsTest(
-    name: String,
-    isLive: Boolean,
-    variants: List[AmountsTestVariant],
-    seed: Int,
+  testName: String,
+  liveTestName: Option[String],
+  isLive: Boolean,
+  target: String,
```

Comment:
It seems a shame to define this as a string - would it make things nicer for SDC if this type encoded whether it's a region or a country? In the old scoping doc I suggested something like `{ region: Region } | { countryCode: string }` for the TS

Reply:
I don't understand this comment as this is a Scala file and the comment talks about TS definitions.

In TS, the target field is typed as `Territory` which is defined in the `src/utils/models.ts` file as follows:

<img width="545" alt="Screenshot 2023-07-24 at 13 27 55" src="https://github.com/guardian/support-admin-console/assets/5357530/5a3b4a07-b62c-4e67-b027-b0bb5daf0ae7">

The `app/models/Region.scala` file defines a `Region` object but I don't know how to add code to that file to create a type that would be either a Region, or a 2-letter string:

<img width="567" alt="Screenshot 2023-07-24 at 13 33 45" src="https://github.com/guardian/support-admin-console/assets/5357530/50ce5831-56e1-4a13-81ed-351c266b20ce">

Reply:
My comment is more general - whatever we do, the scala + TS models would have to align.
This field can hold two different types of thing, but currently the model doesn't convey which it is - it just ends up as a string in dynamodb. I'm wondering if it's better for the apps which read this data if they can check the type. A model like `{ region: Region } | { countryCode: string }` would allow this.

Reply:
If we were to use the above TS model, I think the scala would look something like this:

```
sealed trait AmountsTestTargeting
case class RegionTargeting(region: Region) extends AmountsTestTargeting
case class CountryTargeting(countryCode: String) extends AmountsTestTargeting
```
And for decoding it:
```
object AmountsTestTargeting {
  import io.circe.generic.auto._
  import cats.syntax.functor._  // for the widen syntax

  implicit val amountsTestTargetingDecoder: Decoder[AmountsTestTargeting] =
    Decoder[RegionTargeting].widen or Decoder[CountryTargeting].widen
}
```

Reply:
I agree with the above comments. If we can it'd be really nice to have the type system convey what type of geographical entity we're talking about as well as enforcing, whether it's a region or country code, that a valid value is passed (e.g. right now the string "foo" would keep the type system happy, even though it's neither a country code or a region).

Reply:
Any thoughts on this one?